lua/wikis/commons/Infobox/League.lua

```diff
@@ -284,13 +285,38 @@ end
 function League:customParseArguments(args)
 end
 
+function League:_tournamentPhaseCategory()
+	local startTimestamp = TournamentService.parseDateRecord(self.data.startDate or self.data.endDate)
+	local endTimestamp = TournamentService.parseDateRecord(self.data.endDate)
+
+	if self.data.status == 'finished' then
+		return 'Finished Tournaments'
+	end
+	if not startTimestamp then
+		return 'Upcoming Tournaments'
+	end
+	if DateExt.getCurrentTimestamp() < startTimestamp.timestamp then
+		return 'Upcoming Tournaments'
+	end
+	if not endTimestamp then
+		return 'Live Tournaments'
+	end
+	local oneDayInSeconds = 24 * 60 * 60
+	if DateExt.getCurrentTimestamp() < (endTimestamp.timestamp + oneDayInSeconds) then
+		return 'Live Tournaments'
+	end
+	return 'Finished Tournaments'
```

Comment:
Could do something like
```suggestion
    local phaseMapping = {
        ONGOING = 'Live Tournaments',
        UPCOMING = 'Upcoming Tournaments',
        FINISHED = 'Finished Tournaments'
    }
    
    return phaseMapping[TournamentService.calculatePhase(self.data)]
```
(probably need to make a table to pass instead of data)

Reply:
In my head, I would need to refactor a bunch of existing code to make it a clean solution. I could build a tournament looking data structure, but I think that's just error prone in the future without a proper solution. 

So in this case I feel either a proper refactor (complete of the how we deal with data in at least this infobox, including on all wikis), or this duplicate solution, are the 2 ways forward. 

But maybe there's a smaller tweak that I'm missing?